parsing/dialects/intermediate/dialects/dialect1/src/test/java/org/apache/calcite/test/Dialect1ParserTest.java

```diff
@@ -5591,4 +5591,44 @@ private void checkExNotNull(SqlParserUtil.StringAndPos sap, Throwable thrown) {
         + "END";
     sql(sql).ok(expected);
   }
+
+  @Test public void testLeaveStmtInBeginEndClause() {
+    final String sql = "create procedure foo ()\n"
+        + "label1: begin\n"
+        + "leave label1;\n"
+        + "end label1";
+    final String expected = "CREATE PROCEDURE `FOO` ()\n"
+        + "`LABEL1`: BEGIN\n"
+        + "LEAVE `LABEL1`;\n"
+        + "END `LABEL1`";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testLeaveStmtInIterationStmt() {
+    final String sql = "create procedure foo (bar integer) "
+        + "label1: "
+        + "while bar = 1 do "
+        + "leave label1; "
+        + "end while "
+        + "label1";
+    final String expected =
+        "CREATE PROCEDURE `FOO` (IN `BAR` INTEGER)\n"
+            + "`LABEL1` : WHILE (`BAR` = 1) DO LEAVE `LABEL1`;\n"
+            + "END WHILE `LABEL1`";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testIterateStmtInIterationStmt() {
```

Comment:
Can we also have a test for a leave statement in nested WHILE statements?

Reply:
Test added in https://github.com/googleinterns/calcite/pull/294/commits/5fc12b8c8f66249a38f9627415b7590857921618